### DISCUSSION TITLE
Where do we want to put the timers?

### DISCUSSION THREAD

**Discussion by gnawin** [TulipaEnergy/TulipaEnergyModel.jl#525] [Ideas]
We have two timers both in `run_scenario`. One is using `TimerOutput()`, which shows the time and memory usage and gives a nice overview. The other is simply using `@elapsed` for each code block, and then adding the times in `EnergyProblem`. 

This was done because I assumed people would always use `run_scenario`. However, sometimes we (developers mainly) may want to manually do the steps to run a model. By doing that, there is no information for the time, i.e., an `EnergyProblem` will display 'NaN` for the time taken for the different steps. 

Q1: shall we move the `@elapsed` to inside of each of the function, so you can actually make use of the display?

The other question is about `TimerOutput()`. I kind of think that this is for non-developers (because we want that nice overview), and that they would always use `run_scenario`, so there is no need to move. I can also see that for consistency reasons, maybe we should move that as well, but then it maybe it is not necessary. 

Q2: where do we keep `TimerOutput()`? 

@datejada @abelsiqueira what do you think?

**Comment by clizbe** [comment]
@gnawin I think this has been fixed. Can you confirm and close the discussion if that's true?

**Reply by gnawin** [reply]
Thanks. I also think so.
